src/content_mapblock.cpp

```diff
@@ -1513,7 +1513,50 @@ void mapblock_mesh_generate_special(MeshMakeData *data,
 
 			v3f pos = intToFloat(p, BS);
 
-			std::vector<aabb3f> boxes = n.getNodeBoxes(nodedef);
+			u8 neighbors = 0;
+
+			// locate possible neighboring nodes to connect to
+			if (f.node_box.type == NODEBOX_CONNECTED) {
```

Comment:
i think you should create a function for this condition . This condition is very similar to condition added in collisionMoveResult collisionMoveSimple